swift/apple/FirezoneKit/Sources/FirezoneKit/Helpers/Log.swift

```diff
@@ -129,9 +129,12 @@ private final class LogWriter {
 
     workQueue.async {
       do {
-        try jsonData.write(to: self.logFileURL, options: .atomic)
+        let handle = try FileHandle(forWritingTo: self.logFileURL)
+        try handle.seekToEnd()
+        handle.write(jsonData)
+        try handle.close()
```

Comment:
Isn't opening and closing the handle on every write quite expensive?

Reply:
Hmm I'll check into this first thing tomorrow. We don't write very many logs on the swift side so thought it would be ok, because we don't have atomic guarantees with this API.

But since we use a queue to write and the instance is static I don't think there would any thread issues.

Reply:
Fixed

Reply:
Yeah, those were my suspicions, so it's probably not worth optimizing too much. For slower and/or busy disks I'm sure it's a lot more. Log writing is not critical (we can afford to lose or delay a few) so rather than contend I suppose this design is better.